src/main/java/pl/multiplex/multiplexreservation/manager/ReservationManager.java

```diff
@@ -1,4 +1,163 @@
 package pl.multiplex.multiplexreservation.manager;
 
-public class ReservationManager {
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+import pl.multiplex.multiplexreservation.dao.*;
+import pl.multiplex.multiplexreservation.dao.entity.*;
+
+import java.util.*;
+
+@Service
+public class ReservationManager
+{
+
+    private ReservationRepo reservationRepo;
+    private ScreeningRepo screeningRepo;
+    private MovieRepo movieRepo;
+    private SeatRepo seatRepo;
+    private UserRepo userRepo;
+
+    @Autowired
+    public ReservationManager(ReservationRepo reservationRepo, ScreeningRepo screeningRepo, MovieRepo movieRepo, SeatRepo seatRepo, UserRepo userRepo) {
+        this.reservationRepo = reservationRepo;
+        this.screeningRepo = screeningRepo;
+        this.movieRepo = movieRepo;
+        this.seatRepo = seatRepo;
+        this.userRepo = userRepo;
+    }
+
+
+
+
+    public Iterable<Screening> showAll() {return screeningRepo.findAll();}
+
+    public List<Screening> showByTimeAndDay(int date, int time){
+        return screeningRepo.findScreeningByTimeAndDate(date,time);
+    }
+    public List<String> showByTimeAndMinMaxDayOnlyTitle(int date, int timeMin, int timeMax) {
+        List<Screening> availableScreenings =  screeningRepo.findScreeningByMinMaxTimeAndDate(date, timeMin, timeMax);
+        Set<String> availableMovies = new HashSet<String>();
+        List<String> moviesList = new LinkedList<String>();
+        Screening scr;
+        Iterator screenIter = availableScreenings.iterator();
+        while(screenIter.hasNext()){
+            scr = (Screening) screenIter.next();
+            if(!availableMovies.contains(scr.getMovieId())){
+                availableMovies.add(scr.getMovieId().getTitle());
+            }
+        }
+
+        if(!availableMovies.isEmpty()){
+            Iterator<String> moviesIter = availableMovies.iterator();
+            while(moviesIter.hasNext()){
+                moviesList.add(moviesIter.next());
+            }
+        }
+
+        Collections.sort(moviesList);
```

Comment:
Jak to cudo się zachowa, jeśli wrzucimy tytuły z polskimi znakami?

Reply:
Obawiam się, że ta aplikacja nie obsługuje polskich znaków.